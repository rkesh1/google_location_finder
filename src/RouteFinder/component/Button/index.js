import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

const Button = (props) => {
  const { value, clickHandler, additionClass, disabled } = props;
  return (
    <button type="button" onClick={clickHandler} className={additionClass} disabled={disabled}>{value}</button>
  );
};

Button.propTypes = {
  value: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
  additionClass: PropTypes.string,
  disabled: PropTypes.bool,
};

Button.defaultProps = {
  additionClass: '',
  disabled: false
};
export default Button;
