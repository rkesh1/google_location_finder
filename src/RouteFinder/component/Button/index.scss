button{
    background: white;
    margin-right: 20px;
    padding: 10px 25px;
    &.primary-btn{
        background-color: green;
    }
}
