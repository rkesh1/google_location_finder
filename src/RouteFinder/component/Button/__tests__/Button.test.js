import React from 'react';
import { shallow } from 'enzyme';
import Button from '../index';


describe('<Button /> Component Test Case ', () => {
  let wrapper;
  const props = {
    value: 'Some Value',
    clickHandler: () => { },
    additinalClass: '',
  };
  /**
   * Renders a shallow version of the Button
   */
  function renderShallow() {
    wrapper = shallow(<Button {...props} />);
  }


  it('Button Component render', () => {
    renderShallow();
    expect(wrapper).toMatchSnapshot();
  });

});
