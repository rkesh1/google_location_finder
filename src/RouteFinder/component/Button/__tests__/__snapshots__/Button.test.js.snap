// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Component Test Case  Button Component render 1`] = `ShallowWrapper {}`;
